apps/remote_control/test/lexical/remote_control/code_mod/ast/aliases_test.exs

```diff
@@ -0,0 +1,318 @@
+
+    test "multiple aliases leading by current module" do
+      aliases =
+        ~q[
+          defmodule TopLevel do
+            alias __MODULE__.{First, Second}
+            |
+          end
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:First] == TopLevel.First
+      assert aliases[:Second] == TopLevel.Second
+    end
+
+    test "multiple aliases leading by current module's child" do
+      aliases =
+        ~q[
+          defmodule TopLevel do
+            alias __MODULE__.Child.{First, Second}
+            |
+          end
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:First] == TopLevel.Bar.First
+      assert aliases[:Second] == TopLevel.Bar.Second
+    end
+
+    test "aliases expanding other aliases" do
+      aliases =
+        ~q[
+          alias Foo.Bar.Baz
+          alias Baz.Quux|
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:Baz] == Foo.Bar.Baz
+      assert aliases[:Quux] == Foo.Bar.Baz.Quux
+    end
+
+    test "aliases expanding current module" do
+      aliases = ~q[
+        defmodule TopLevel do
+          alias __MODULE__.Foo|
+        end
+      ] |> aliases_at_cursor()
+
+      assert aliases[:Foo] == TopLevel.Foo
+    end
+
+    test "allows overrides" do
+      aliases =
+        ~q[
+          alias Foo.Bar.Baz
+          alias Other.Baz
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:Baz] == Other.Baz
+    end
+  end
+
+  describe "nested modules" do
+    test "no aliases are defined for modules with dots" do
+      aliases =
+        ~q[
+          defmodule GrandParent.Parent.Child do
+           |
+          end
+        ]
+        |> aliases_at_cursor()
+
+      refute Map.has_key?(aliases, :Child)
+    end
+
+    test "with children get their parents name" do
+      aliases =
+        ~q[
+          defmodule Grandparent.Parent do
+            defmodule Child do
+              |
+            end
+          end
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:Child] == Grandparent.Parent.Child
+      assert aliases[:__MODULE__] == Grandparent.Parent.Child
+    end
+  end
+
+  describe "alias scopes" do
+    test "aliases are removed when leaving a module" do
+      aliases =
+        ~q[
+          defmodule Basic do
+            alias Foo.Bar
+          end|
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases == %{Basic: Basic, __MODULE__: nil}
+    end
+
+    test "aliases inside of nested modules" do
+      aliases =
+        ~q[
+          defmodule Parent do
+            alias Foo.Grandparent
+
+            defmodule Child do
+              alias Foo.Something
+              |
+            end
+          end
+        ]
+        |> aliases_at_cursor()
+
+      assert aliases[:Grandparent] == Foo.Grandparent
+      assert aliases[:__MODULE__] == Parent.Child
+      assert aliases[:Something] == Foo.Something
+    end
+
+    test "a nested module is aliased inside the module" do
```

Comment:
since there is already a `describe` about nested module, I think we can delete this one right now.